src/instructlab/model/download.py

```diff
@@ -82,20 +84,18 @@ def download(self):
             )
 
         try:
-            if self.ctx.obj is not None:
-                hf_logging.set_verbosity(self.ctx.obj.config.general.log_level.upper())
+            if self.log_level is not None:
+                hf_logging.set_verbosity(self.log_level)
             files = list_repo_files(repo_id=self.repository, token=self.hf_token)
             if any(re.search(r"\.(safetensors|bin)$", fname) for fname in files):
                 self.download_entire_hf_repo()
             else:
                 self.download_gguf()
 
         except Exception as exc:
-            click.secho(
-                f"\nDownloading model failed with the following Hugging Face Hub error:\n{DEFAULT_INDENT}{exc}",
-                fg="red",
-            )
-            raise click.exceptions.Exit(1)
+            raise ValueError(
```

Comment:
Why a `ValueError`?

Reply:
that's just what ive been defaulting to
if there is a specific error I should use Idm switching to that

Reply:
`ValueError` suggests something specific, not sure if that applies to all the usage I'm seeing here: https://docs.python.org/3/library/exceptions.html#ValueError

I've defined custom exceptions for `upload` https://github.com/instructlab/instructlab/blob/main/src/instructlab/model/upload.py#L20-L51

Reply:
fair enough, ive updated most of the calls with something more appropriate
custom errors seems like overkill to me

Reply:
I think custom errors should be introduced given this will be the basis for a true API, but since this is a refactor PR I won't block on this